ecl/hql/hqlgram2.cpp

```diff
@@ -1404,11 +1440,9 @@ void HqlGram::addAssignment(attribute & target, attribute &source)
     }
     else if (targetOp == no_select)
     {
-        // self.* := expr;      assertex(targetExpr->getOperator()==no_select);
-        if (findAssignment(targetExpr))
+        // self.* := expr;
+        if (checkAlreadyAssigned(target, targetExpr))
```

Comment:
If checkAlreadyAssigned produces a warning message, the assignment will no longer be made.  Is that ok?